### ISSUE TITLE
Unable to run "Hello World" example on macOS Darwin 17.7.0

### ISSUE THREAD

**Issue by davidsnyder** [movitto/reterm#1]
Hi,

I'm trying to use reterm but I'm getting errors when running the hello world example that collects input. 
Here's the code: https://gist.github.com/davidsnyder/23bfbb4533f66f951617f76fde1bd1c9

My Gemfile has reterm 0.6.3, and ncursesw 1.4.10. I installed ncurses and cdk on my mac using the brew package manager. 

When I run the above program, I get: 

```
$ bundle exec ruby reterm.rb
Unhandled error:
  undefined method `id' for nil:NilClass
    /Library/Ruby/Gems/2.3.0/gems/reterm-0.6.3/lib/reterm/window.rb:495:in `colors='
    reterm.rb:8:in `block in <main>'
    /Library/Ruby/Gems/2.3.0/gems/reterm-0.6.3/lib/reterm/init.rb:53:in `init_reterm'
    reterm.rb:6:in `<main>'
Input results:
{}
```

I commented out   `win.colors = :main` to try to get past this and got 

```
$ bundle exec ruby reterm.rb
Unhandled error:
  undefined method `sum' for []:Array
    /Library/Ruby/Gems/2.3.0/gems/reterm-0.6.3/lib/reterm/layouts/horizontal.rb:12:in `current_cols'
    /Library/Ruby/Gems/2.3.0/gems/reterm-0.6.3/lib/reterm/layout.rb:66:in `exceeds_bounds?'
    /Library/Ruby/Gems/2.3.0/gems/reterm-0.6.3/lib/reterm/layout.rb:149:in `add_child'
    /Library/Ruby/Gems/2.3.0/gems/reterm-0.6.3/lib/reterm/layouts/horizontal.rb:30:in `add_child'
    reterm.rb:14:in `block in <main>'
    /Library/Ruby/Gems/2.3.0/gems/reterm-0.6.3/lib/reterm/init.rb:53:in `init_reterm'
    reterm.rb:6:in `<main>'
Input results:
{}
```

At this point I'm not sure what to do. Perhaps I am missing some ancillary libraries around ncurses. Your RedHat/Fedora instructions specify some additional libraries, like ncurses-devel, which I haven't been able to find for Darwin. 

Any help is appreciated,

David

**Comment by movitto** [comment]
Hi David, glad to help.

The first error is due to the fact that colors are not initialized before you try to assign them. In your logic you're assigning the "main" color scheme to the created window, but no such color scheme exists (admitadly this could use a better error). If you simply add the following line of code in the **init_reterm** block before you assign the colors, it will work:

```
init_reterm {
  ColorPair.register :green, :black, :main

  win = Window.new
  win.colors = :main
  win.border!
```

See [ui.rb](https://github.com/movitto/reterm/blob/master/examples/ui.rb#L8) for an example of this as well.

After I made that change I was able to run the provided example successfully. As far as your second error, it seems Ruby did not introduce **sum** into the **Enumerable** class (which Array derives from) until [2.4](https://blog.bigbinary.com/2016/11/02/ruby-2-4-introduces-enumerable-sum.html) (I see from your gem path you are using Ruby 2.3)

Would it be possible to use Ruby 2.4.0 or later? If not perhaps we can monkey patch it into Array ourselves with a conditional only applying it for Ruby < 2.4

**Comment by davidsnyder** [comment]
Thank you! That's exactly right. Should the README be updated to include that line?

I also updated my Ruby version. I feel silly for not catching that.

I was then able to run the example, but upon clicking in the UI the program crashed again with a different error: https://gist.github.com/davidsnyder/c800efcf2a9ebf3a673c8f5f93f35c76

Maybe I'm missing something here too (maybe I'm supposed to register something before I can handle mouse input) but I think it's a bad experience to have the program crash without a clear error.

**Comment by movitto** [comment]
@davidsnyder I just pushed three commits to the repo.

The first updates the README to include the fix.

The second fixes the error you reported w/ nav_select. It's been a while since I worked on this library so I'm hoping it doesn't affect anything else (I noticed that even with this mouse navigation in other example wasn't working 100% as intended, will have to look into this in the future; unless you wanted to get to it first!)

The third patch monkey patches Enumerable#sum in for Ruby < 2.4. Unfortunately I don't have a 2.3 environment at hand, so you or someone else will have to test it out and report if it works there. I confirmed that my Ruby 2.5 env was unaffected by this.

Hope it helps!
